parts/k8s/cloud-init/artifacts/cis.sh

```diff
@@ -4,8 +4,15 @@ assignRootPW() {
   if grep '^root:[!*]:' /etc/shadow; then
     SALT=$(openssl rand -base64 5)
     SECRET=$(openssl rand -base64 37)
-    CMD="import crypt, getpass, pwd; print crypt.crypt('$SECRET', '\$6\$$SALT\$')"
-    HASH=$(python -c "$CMD")
+    if command -v python; then
+      CMD="import crypt, getpass, pwd; print crypt.crypt('$SECRET', '\$6\$$SALT\$')"
```

Comment:
Couldn't the CMD value be set above the if clause?  That way, since it is the same for both branches, you don't have it duplicated (and potentially drift)?

Not critical but it would simplify things slightly and reduce the size of the script by that one line.

Reply:
The interface for the `print` builtin (or equivalent) changed in python v3 to require a paren, so both `CMD` and `HASH` are different depending on the version.

Agree that there are probably further efficiencies possible here and in this change altogether, but altogether agree that based on the aks-engine EOL velocity, the priority is just to get something working ASAP (which this current implementation does — see the VHD pipeline outcomes) so we can start smoke-testing a 20.04-LTS VHD before 18.04-LTS is officially distro non grata.